Makefile

```diff
@@ -170,20 +170,42 @@ PROTO_SRC_NAMES = $(basename $(notdir $(PROTO_SRCS)))
 PROTO_GO_OUTS = $(foreach name, $(PROTO_SRC_NAMES), go/vt/proto/$(name)/$(name).pb.go)
 
 # This rule rebuilds all the go files from the proto definitions for gRPC.
-proto: $(PROTO_GO_OUTS)
-
-ifndef NOBANNER
-	echo $$(date): Compiling proto definitions
-endif
+PROTOC = /tmp/vitess-make-protoc
+
+install_protoc-gen-go-3-12-4:
+	rm -f ${PROTOC}
+	if [ -f ${VTROOT}/bin/protoc ] && ${VTROOT}/bin/protoc --version | grep -q "3.12.4" ; then \
+		ln -s ${VTROOT}/bin/protoc ${PROTOC}
+	elif /tmp/proto-builder/protoc/bin/protoc --version | grep -q "3.12.4" ; then \
+		ln -s "/tmp/proto-builder/protoc/bin/protoc" ${PROTOC}
+		echo "OK"
+	elif ! ${PROTOC} --version | grep -q "3.12.4" ; then \
+		echo "getting protoc..."
+		PROTOC_GEN_GO_VERSION=v1.3.2
+		rm -rf "/tmp/proto-builder" && \
+		mkdir -p /tmp/proto-builder && \
+    cd /tmp/proto-builder && \
+    go mod init proto-builder && \
+    GO111MODULE=on go get golang.org/x/tools/cmd/goimports github.com/golang/protobuf/protoc-gen-go@$${PROTOC_GEN_GO_VERSION} && \
+		PROTOC_VERSION=3.12.4 && \
+		curl -LO https://github.com/protocolbuffers/protobuf/releases/download/v$${PROTOC_VERSION}/protoc-$${PROTOC_VERSION}-linux-x86_64.zip && \
+    unzip protoc-$${PROTOC_VERSION}-linux-x86_64.zip -d ./protoc && \
+		ln -s "/tmp/proto-builder/protoc/bin/protoc" ${PROTOC}
+	fi
 
-$(PROTO_GO_OUTS): install_protoc-gen-go proto/*.proto
+proto:
+	install_protoc-gen-go-3-12-4 proto/*.proto
```

Comment:
What system are you running this on? On macos Catalina (which ships with Make 3.81) it fails with:

```
❯ make proto
/bin/bash: install_protoc-gen-go-3-12-4: command not found
make: *** [proto] Error 127
```

Reply:
Hmmm OK; I'm running on Ubuntu Linux. I'll look into it.